torch/csrc/jit/script/init.cpp

```diff
@@ -111,6 +111,27 @@ struct PythonResolver : public Resolver {
     }
 
     if (isNamedTupleClass(obj)) {
+      // Currently don't support default values
+      if (py::hasattr(obj, "_field_defaults")) {
+        auto default_dict = py::cast<std::map<std::string, py::object>>(
+            py::getattr(obj, "_field_defaults"));
+        if (default_dict.size()) {
+          std::string error_msg =
+              "Default values are currently not supported"
+              " on NamedTuple fields in TorchScript. Fields "
+              "with default values: [";
+          bool first = true;
+          for (const auto& kv : default_dict) {
+            if (!first) {
+              error_msg += ", ";
+            }
+            error_msg += kv.first;
+          }
+          error_msg += "]";
+          throw std::runtime_error(error_msg);
```

Comment:
I would recommend trying to communicate this in any way but throwing a generic exception. For instance, it is probably better to just pass SourceRange to resolveType and let this report the full error message.